pkg/ovsdb/transact_test.go

```diff
@@ -1042,24 +1042,143 @@ func TestTransactDelete(t *testing.T) {
 	assert.False(t, ok)
 }
 
-func TestTransactWait(t *testing.T) {
+func TestTransactWaitEQ(t *testing.T) {
 	table := "table1"
 	timeout := 0
+	row1 := map[string]interface{}{
+		"key1": "val1a",
+		"key2": 1,
+	}
+	row2 := map[string]interface{}{
+		"key1": "val1b",
+		"key2": 1,
+	}
+	columns := []string{"key1"}
+	rows := []map[string]interface{}{
+		{
+			"key1": "val1a",
+		},
+	}
+	until := FN_EQ
 	req := &libovsdb.Transact{
 		DBName: "simple",
 		Operations: []libovsdb.Operation{
+			{
+				Op:      OP_INSERT,
+				Table:   &table,
+				Row:     &row1,
+				Timeout: &timeout,
+			},
+			{
+				Op:      OP_INSERT,
+				Table:   &table,
+				Row:     &row2,
+				Timeout: &timeout,
+			},
 			{
 				Op:      OP_WAIT,
 				Table:   &table,
+				Rows:    &rows,
+				Columns: &columns,
+				Until:   &until,
 				Timeout: &timeout,
 			},
 		},
 	}
 	common.SetPrefix("ovsdb/nb")
+	testEtcdCleanup(t)
 	resp, _ := testTransact(t, req)
 	assert.Nil(t, resp.Error)
 }
 
+func TestTransactWaitNE(t *testing.T) {
+	table := "table1"
+	timeout := 0
+	row1 := map[string]interface{}{
+		"key1": "val1a",
+		"key2": 1,
+	}
+	row2 := map[string]interface{}{
+		"key1": "val1b",
+		"key2": 1,
+	}
+	columns := []string{"key1"}
+	rows := []map[string]interface{}{
+		{
+			"key1": "val1c",
+		},
+	}
+	until := FN_NE
+	req := &libovsdb.Transact{
+		DBName: "simple",
+		Operations: []libovsdb.Operation{
+			{
+				Op:      OP_INSERT,
+				Table:   &table,
+				Row:     &row1,
+				Timeout: &timeout,
+			},
+			{
+				Op:      OP_INSERT,
+				Table:   &table,
+				Row:     &row2,
+				Timeout: &timeout,
+			},
+			{
+				Op:      OP_WAIT,
+				Table:   &table,
+				Rows:    &rows,
+				Columns: &columns,
+				Until:   &until,
+				Timeout: &timeout,
+			},
+		},
+	}
+	common.SetPrefix("ovsdb/nb")
+	testEtcdCleanup(t)
+	resp, _ := testTransact(t, req)
+	assert.Nil(t, resp.Error)
+}
+
+func TestTransactWaitError(t *testing.T) {
```

Comment:
related to my comment about equal is "false", you don't have a test where until is FN_NE, but the rows are equals.

Reply:
agree, added